src/serialize.rs

```diff
@@ -0,0 +1,397 @@
+//! Convenience tools for serializing [`World`]s; requires the `serde` feature
+//!
+//! [`Component`]s are not necessarily serializable, so we cannot directly implement [`Serialize`]
+//! for [`World`]. The helper functions defined in this module allow serialization and
+//! deserialization to rely on purpose-defined traits to control the procedures explicitly.
+//!
+//! These tools use public hecs APIs, and are somewhat opinionated. For some applications, a more
+//! custom approach may be preferable.
```

Comment:
An extra sentence here pointing towards the API's for implementing that custom approach might be useful.